File: android/worker/app/src/main/java/com/workforcex/worker/ui/worker/WorkerDashboardActivity.java
package com.workforcex.worker.ui.worker;

import android.content.Intent;
import android.os.Bundle;
import android.view.Menu;
import android.view.MenuItem;
import android.view.View;
import android.widget.TextView;
import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import com.workforcex.worker.R;
import com.workforcex.worker.api.Notification;
import com.workforcex.worker.api.RetrofitClient;
import com.workforcex.worker.api.WorkerProfileResponse;
import com.workforcex.worker.databinding.ActivityWorkerDashboardBinding;
import com.workforcex.worker.utils.TokenManager;
import java.util.List;
import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class WorkerDashboardActivity extends AppCompatActivity {

    private ActivityWorkerDashboardBinding binding;
    private TokenManager tokenManager;
    private TextView notificationBadge;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityWorkerDashboardBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());
        tokenManager = new TokenManager(this);

        setSupportActionBar(binding.toolbar);
        setTitle("Worker Dashboard");

        binding.btnBrowseJobs.setOnClickListener(v -> startActivity(new Intent(this, BrowseJobsActivity.class)));
        binding.btnMyApplications.setOnClickListener(v -> startActivity(new Intent(this, MyApplicationsActivity.class)));
        binding.btnEditProfile.setOnClickListener(v -> startActivity(new Intent(this, WorkerProfileActivity.class)));
        binding.btnVerification.setOnClickListener(v -> startActivity(new Intent(this, VerificationActivity.class)));
        binding.btnLogout.setOnClickListener(v -> {
            tokenManager.clear();
            Intent intent = new Intent(this, com.workforcex.worker.ui.auth.LoginActivity.class);
            intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TASK);
            startActivity(intent);
        });
    }

    @Override
    protected void onResume() {
        super.onResume();
        loadProfile();
        loadNotificationCount();
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(R.menu.dashboard_menu, menu);
        final MenuItem menuItem = menu.findItem(R.id.action_notifications);
        View actionView = menuItem.getActionView();
        notificationBadge = actionView.findViewById(R.id.notification_badge);
        actionView.setOnClickListener(v -> onOptionsItemSelected(menuItem));
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(@NonNull MenuItem item) {
        if (item.getItemId() == R.id.action_notifications) {
            startActivity(new Intent(this, NotificationsActivity.class));
            return true;
        }
        return super.onOptionsItemSelected(item);
    }

    private void loadProfile() {
        RetrofitClient.get().getWorkerProfile(tokenManager.getBearerToken())
                .enqueue(new Callback<WorkerProfileResponse>() {
                    @Override
                    public void onResponse(Call<WorkerProfileResponse> call, Response<WorkerProfileResponse> response) {
                        if (response.isSuccessful() && response.body() != null) {
                            WorkerProfileResponse p = response.body();
                            binding.tvWelcome.setText("Welcome, " + (p.name != null ? p.name : tokenManager.getMobile()));
                            binding.tvSkills.setText("Skills: " + (p.skills != null ? p.skills : "Not set"));
                            binding.tvExperience.setText("Experience: " + (p.experience != null ? p.experience + " yrs" : "Not set"));
                        }
                    }
                    @Override
                    public void onFailure(Call<WorkerProfileResponse> call, Throwable t) {}
                });
    }

    private void loadNotificationCount() {
        RetrofitClient.get().getUnreadNotifications(tokenManager.getBearerToken())
                .enqueue(new Callback<List<Notification>>() {
                    @Override
                    public void onResponse(Call<List<Notification>> call, Response<List<Notification>> response) {
                        if (response.isSuccessful() && response.body() != null && notificationBadge != null) {
                            int count = response.body().size();
                            if (count > 0) {
                                notificationBadge.setText(String.valueOf(count));
                                notificationBadge.setVisibility(View.VISIBLE);
                            } else {
                                notificationBadge.setVisibility(View.GONE);
                            }
                        }
                    }
                    @Override
                    public void onFailure(Call<List<Notification>> call, Throwable t) {}
                });
    }
}

File: android/worker/app/src/main/java/com/workforcex/worker/ui/worker/WorkerDashboardActivity.kt
package com.workforcex.worker.ui.worker

import android.content.Intent
import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import android.view.View
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import com.workforcex.worker.R
import com.workforcex.shared.models.Notification
import com.workforcex.shared.RetrofitClient
import com.workforcex.shared.models.WorkerProfileResponse
import com.workforcex.worker.databinding.ActivityWorkerDashboardBinding
import com.workforcex.worker.ui.auth.LoginActivity
import com.workforcex.worker.utils.TokenManager
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

class WorkerDashboardActivity : AppCompatActivity() {

    private lateinit var binding: ActivityWorkerDashboardBinding
    private lateinit var tokenManager: TokenManager
    private var notificationBadge: TextView? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityWorkerDashboardBinding.inflate(layoutInflater)
        setContentView(binding.root)
        tokenManager = TokenManager(this)

        setSupportActionBar(binding.toolbar)
        title = "Worker Dashboard"

        binding.btnBrowseJobs.setOnClickListener { startActivity(Intent(this, BrowseJobsActivity::class.java)) }
        binding.btnMyApplications.setOnClickListener { startActivity(Intent(this, MyApplicationsActivity::class.java)) }
        binding.btnEditProfile.setOnClickListener { startActivity(Intent(this, WorkerProfileActivity::class.java)) }
        binding.btnVerification.setOnClickListener { startActivity(Intent(this, VerificationActivity::class.java)) }
        binding.btnLogout.setOnClickListener {
            tokenManager.clear()
            val intent = Intent(this, LoginActivity::class.java)
            intent.flags = Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_CLEAR_TASK
            startActivity(intent)
        }
    }

    override fun onResume() {
        super.onResume()
        loadProfile()
        loadNotificationCount()
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        menuInflater.inflate(R.menu.dashboard_menu, menu)
        val menuItem = menu.findItem(R.id.action_notifications)
        val actionView = menuItem.actionView
        notificationBadge = actionView?.findViewById(R.id.notification_badge)
        actionView?.setOnClickListener { onOptionsItemSelected(menuItem) }
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        if (item.itemId == R.id.action_notifications) {
            startActivity(Intent(this, NotificationsActivity::class.java))
            return true
        }
        return super.onOptionsItemSelected(item)
    }

    private fun loadProfile() {
        RetrofitClient.get().getWorkerProfile(tokenManager.getBearerToken())
            .enqueue(object : Callback<WorkerProfileResponse> {
                override fun onResponse(
                    call: Call<WorkerProfileResponse>,
                    response: Response<WorkerProfileResponse>
                ) {
                    if (response.isSuccessful && response.body() != null) {
                        val p = response.body()!!
                        binding.tvWelcome.text = "Welcome, " + (p.name ?: tokenManager.getMobile())
                        binding.tvSkills.text = "Skills: " + (p.skills ?: "Not set")
                        binding.tvExperience.text = "Experience: " + (p.experience?.let { "$it yrs" } ?: "Not set")
                    }
                }

                override fun onFailure(call: Call<WorkerProfileResponse>, t: Throwable) {}
            })
    }

    private fun loadNotificationCount() {
        RetrofitClient.get().getUnreadNotifications(tokenManager.getBearerToken())
            .enqueue(object : Callback<List<Notification>> {
                override fun onResponse(
                    call: Call<List<Notification>>,
                    response: Response<List<Notification>>
                ) {
                    if (response.isSuccessful && response.body() != null && notificationBadge != null) {
                        val count = response.body()!!.size
                        if (count > 0) {
                            notificationBadge!!.text = count.toString()
                            notificationBadge!!.visibility = View.VISIBLE
                        } else {
                            notificationBadge!!.visibility = View.GONE
                        }
                    }
                }

                override fun onFailure(call: Call<List<Notification>>, t: Throwable) {}
            })
    }
}